automated scenario for service specified experiment

kalantar opened the pull request:
automated e2e scenario for v0.2 -- service centric experiment specification. Addresses issue https://github.com/iter8-tools/iter8-controller/issues/180.

kalantar commented:
This will probably fail testing. It is written in the expectation of upcoming controller changes and is dependent on them. See issue https://github.com/iter8-tools/iter8-controller/issues/156.

kalantar commented:
@sriumcp this may be enough to allow you to get started on issue https://github.com/iter8-tools/docs/issues/90

kimikowang commented on doc/tutorials/istio/bookinfo/service/canary_productpage-v1_to_productpage-v2.yaml:
```diff
@@ -0,0 +1,29 @@
+  routingReference:
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    name: bookinfo
```
The vs referenced here does not exist.

kalantar replied:
Agree. My mistake. I've removed this,

kimikowang commented:
What is the reason of creating all the new service/deployments yaml?

kimikowang commented:
And this does not conform to the design doc use case. Please take a look at the targeted user case.

kalantar commented:
> What is the reason of creating all the new service/deployments yaml?

I believe we need to redefine our definition to have different services for each new version and a new gateway definition. For simplicity of the user, I thought it made sense to include all the necessary pieces together. Otherwise, I think we would need to break up the bookinfo-tutorial.yaml file.

kalantar commented:
> What is the reason of creating all the new service/deployments yaml?

@kimikowang After our discussion, I removed the duplication of the yaml and have just replaced where needed and added additional.

kimikowang commented on test/e2e/e2e-scenario-5s.sh:
```diff
@@ -0,0 +1,56 @@
+sleep 1
+kubectl wait --for=condition=ExperimentCompleted -n $NAMESPACE experiments.iter8.tools productpage-v2-rollout --timeout=600s
+kubectl get experiments -n $NAMESPACE
+kubectl get vs bookinfo -n $NAMESPACE -o yaml
```
This should be removed, or `kubectl get vs -n $NAMESPACE -o yaml`.
